include/rdma/fi_endpoint.h

```diff
@@ -208,6 +208,20 @@ fi_getopt(fid_t fid, int level, int optname,
 	return ep->ops->getopt(fid, level, optname, optval, optlen);
 }
 
+static inline int fi_ep_alias(struct fid_ep *ep, struct fid_ep **alias_ep,
+			      uint64_t flags)
+{
+	int ret;
+	struct fid *fid;
+	struct fid_ep *fid_ep;
+	ret = fi_alias(&ep->fid, &fid, flags);
+	if (ret)
+		return ret;
+	fid_ep = container_of(fid, struct fid_ep, fid);
+	*alias_ep = fid_ep;
```

Comment:
Just set alias_ep directly.  I'd also slightly simplify this to:
if (!ret)
    *alias_ep = ...
return ret;